File: GenericTesting/GenericVBTesting/Models/DemandTrendInput.vb
Imports System.Xml.Serialization

<Serializable>
Public Class DemandTrendInput
  <XmlAttribute>
  Public Property FIKey As Integer
  <XmlAttribute>
  Public Property StartDate As Date
  <XmlAttribute>
  Public Property EndDate As Date


  Public Property DemandPlans As List(Of Integer)

  Public Property DemandLocations As List(Of Integer)

  Public Sub New()

  End Sub

  Public Sub New(fiKey As Integer, startDate As Date, endDate As Date, demandPlans As List(Of Integer), demandLocations As List(Of Integer))
    Me.FIKey = fiKey
    Me.StartDate = startDate
    Me.EndDate = endDate
    Me.DemandPlans = demandPlans
    Me.DemandLocations = demandLocations
  End Sub
End Class

File: GenericTesting/GenericTesting/Models/DemandTrendInput.cs
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Xml;
using System.Xml.Schema;
using System.Xml.Serialization;

namespace GenericTesting.Models
{
  [Serializable]
  [XmlRoot]
  public class DemandTrendInput 
  {
    [XmlIgnore]
    public string DemandTrendName { get; set; }
    [XmlAttribute]
    public string Grouping { get; set; }
    [XmlAttribute]
    public int FIKey { get; set; }
    [XmlAttribute]
    public DateTime StartDate { get; set; }
    [XmlAttribute]
    public DateTime EndDate { get; set; }
    public List<int> DemandLocations { get; set; }

    public DemandTrendInput()
    {
    }

    public DemandTrendInput(string demandTrendName, int fiKey, System.DateTime startDate, System.DateTime endDate, TrendChoices grouping, List<int> demandLocations)
    {
      DemandTrendName = demandTrendName;
      FIKey = fiKey;
      StartDate = startDate;
      EndDate = endDate;
      Grouping = grouping.ToString();
      DemandLocations = demandLocations;
    }                               
  }
}
